### ISSUE TITLE
error handling request

### ISSUE THREAD

**Issue by divemasterjm** [briis/securityspy#14]
i show this error showing many times
Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request
    resp = await self._request_handler(request)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_middlewares.py", line 119, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 56, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 18, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 72, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 129, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 129, in handle
    result = await result
  File "/usr/src/homeassistant/homeassistant/components/camera/__init__.py", line 496, in get
    return await self.handle(request, camera)
  File "/usr/src/homeassistant/homeassistant/components/camera/__init__.py", line 513, in handle
    image = await camera.async_camera_image()
  File "/config/custom_components/securityspy/camera.py", line 166, in async_camera_image
    last_image = await self.secspy.get_snapshot_image(self._device_id)
  File "/usr/local/lib/python3.8/site-packages/pysecspy/secspy_server.py", line 202, in get_snapshot_image
    response = await self.req.get(
  File "/usr/local/lib/python3.8/site-packages/aiohttp/client.py", line 544, in _request
    await resp.start(conn)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/client_reqrep.py", line 890, in start
    message, payload = await self._protocol.read()  # type: ignore
  File "/usr/local/lib/python3.8/site-packages/aiohttp/streams.py", line 604, in read
    await self._waiter
aiohttp.client_exceptions.ServerDisconnectedError: Server disconnected

**Comment by briis** [comment]
Could you turn on debugging for this Component, and post the relevant log entries here.

Add this to your `configuration.yaml` file:

```yaml
logger:
  default: error
  logs:
    custom_components.securityspy: debug
```
Restart Home Assistant and try again.

**Comment by divemasterjm** [comment]
2021-04-08 18:11:26 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:11:26 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:11:26 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:11:26 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:11:26 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:11:26 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:11:26 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:11:26 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:13:37 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:13:37 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:13:37 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:13:37 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:13:37 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:13:37 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:13:37 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:13:37 ERROR (stream_worker) [homeassistant.components.stream.worker] Error demuxing stream:
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording
2021-04-08 18:13:49 DEBUG (MainThread) [custom_components.securityspy.switch] Turning off Recording

**Comment by divemasterjm** [comment]
the issue is in this camera constantly is turning off and on recording
i had to exclude  in recorder this entity to write info cause database was always writing events

**Comment by briis** [comment]
There is obviously something wrong with the stream coming from this camera. I will soon release a new version, where you can disable the RTSP stream. It will be interesting to see if this makes the issue go away.

**Comment by divemasterjm** [comment]
i think the demuxing error is not from this camera, as stated in the log witch i didnt copy

**Comment by briis** [comment]
OK, from the log file you posted above, it seems there is an error getting images from a camera. It is not clear exactly what causes this, but it could be network related.

> File "/config/custom_components/securityspy/camera.py", line 166, in async_camera_image
last_image = await self.secspy.get_snapshot_image(self._device_id)
File "/usr/local/lib/python3.8/site-packages/pysecspy/secspy_server.py", line 202, in get_snapshot_image
response = await self.req.get(

**Comment by divemasterjm** [comment]
strange, i have 8 cameras and i only have this problem in one, also ss is not reflecting any issue

**Comment by briis** [comment]
There must be something with that camera, that is different from the others. I must see if I can do anything to capture some more data so we can figure out what is wrong. And you are absolutely sure this camera is setup exactly as the others?
One thing is that they are the same in SecuritySpy, but are there any settings that are different in the camera itself?

I am just trying to figure out why this specific camera acts differently

**Comment by divemasterjm** [comment]
I've been testing and controlling and all cameras have same setup, by the other hand, if you query settings via http, the switch is always on, I don't see the connection between trap stream and setting on the motion record switch

**Comment by divemasterjm** [comment]
Logger: aiohttp.server
Source: custom_components/securityspy/camera.py:166
First occurred: 7:24:28 PM (7 occurrences)
Last logged: 8:47:50 PM

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request
    resp = await self._request_handler(request)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle
    resp = await handler(request)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/web_middlewares.py", line 119, in impl
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 56, in security_filter_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 18, in request_context_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 74, in ban_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 129, in auth_middleware
    return await handler(request)
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 131, in handle
    result = await result
  File "/usr/src/homeassistant/homeassistant/components/camera/__init__.py", line 498, in get
    return await self.handle(request, camera)
  File "/usr/src/homeassistant/homeassistant/components/camera/__init__.py", line 515, in handle
    image = await camera.async_camera_image()
  File "/config/custom_components/securityspy/camera.py", line 166, in async_camera_image
    last_image = await self.secspy.get_snapshot_image(self._device_id)
  File "/usr/local/lib/python3.8/site-packages/pysecspy/secspy_server.py", line 202, in get_snapshot_image
    response = await self.req.get(
  File "/usr/local/lib/python3.8/site-packages/aiohttp/client.py", line 544, in _request
    await resp.start(conn)
  File "/usr/local/lib/python3.8/site-packages/aiohttp/client_reqrep.py", line 890, in start
    message, payload = await self._protocol.read()  # type: ignore
  File "/usr/local/lib/python3.8/site-packages/aiohttp/streams.py", line 604, in read
    await self._waiter
aiohttp.client_exceptions.ServerDisconnectedError: Server disconnected

**Comment by divemasterjm** [comment]
@briis 
ive install 1.0.1 and did not solve the problem disabling RTSP (((
i also try to move camera to another router but same behaviour.
Maybe could be worth not to update so frecruently this switches?

**Comment by briis** [comment]
The switches are updated by the event stream, so they should only get updated if changes occur - that is how it works on my test system, and my logbook shows barely any entries. I will dig some more.

**Comment by divemasterjm** [comment]
if you want me to do some tests just tell me

**Comment by briis** [comment]
Just to see if this could be the area to look at, could you try and change the file `__init__.py` in the `custom_components/securityspy` directory.

Go to line 70, and change the timedelta value to something different than 2, for example 120. So that it looks like this:

`secspy_data = SecuritySpyData(hass, securityspyserver, timedelta(seconds=120))`

**Comment by stbrun** [comment]
Hello

I get the following error messages when I switch the rtsp in the integration:



Best regards, Stefan 

–––––––––––––––––––––––––––––––––––––-

Am 09.04.2021 um 08:09 schrieb divemasterjm ***@***.***>:


@briis <https://github.com/briis>
ive install 1.0.1 and did not solve the problem disabling RTSP (((
i also try to move camera to another router but same behaviour.
Maybe could be worth not to update so frecruently this switches?

**Comment by briis** [comment]
@stbrun 

> I get the following error messages when I switch the rtsp in the integration:

You did not attach the error message

**Comment by stbrun** [comment]
Hello I’m out of order. I can not login again on my HA installation. What can I do?

Stefan
--------------------------------------------------------

Am 09.04.2021 um 08:21 schrieb Bjarne Riis ***@***.***>:


Just to see if this could be the area to look at, could you try and change the file __init__.py in the custom_components/securityspy directory.

Go to line 70, and change the timedelta value to something different than 2, for example 120. So that it looks like this:

secspy_data = SecuritySpyData(hass, securityspyserver, timedelta(seconds=120))

**Comment by divemasterjm** [comment]
@briis same behaviour

Error handling request
8:36:57 AM – (ERROR) custom_components/securityspy/camera.py - message first occurred at 8:30:16 AM and shows up 10 times

**Comment by divemasterjm** [comment]
i dont know if ive told you, only happens with recording_motion switch, others from same camera are ok, for example actions

**Comment by briis** [comment]
Hi @divemasterjm 
I have just made a new Beta release 1.0.2-beta-1. Could you try and install that to see if this has any impact.
If it makes it worse, you can just go back to 1.0.1

**Comment by divemasterjm** [comment]
installing

**Comment by briis** [comment]
I am out of ideas. I will need to let this rest for a while, to see if some useful errors come up. 

The above error has something to do with the `stream` component in Home Assistant, and is not something I can do anything with. I would assume that it will go away if you disable the RTSP stream.

**Comment by divemasterjm** [comment]
thanks @briis, i ve made some more tests and seems more stable.

**Comment by briis** [comment]
Good to hear. If you idon'thave it, could you turn on debug for this Integration as described in the top. Just to see if I can get more data to look at.

**Comment by divemasterjm** [comment]
sure, i will turn on now

**Comment by divemasterjm** [comment]
i dont have anymore the issue, it seems solved

**Comment by briis** [comment]
Sounds good. I will keep this open for a little while to see if it shows up again.

**Comment by divemasterjm** [comment]
@briis thanks for your support

**Comment by briis** [comment]
Closed with V1.0.2
